### ISSUE TITLE
Philips Hue Support

### ISSUE THREAD

**Issue by TJ-91** [psieg/Lightpack#204]
For personal use I've written some code which will sync the Philips Hue behind me with the average color of the screen. Maybe someone here wants to use or improve it and possibly get it far enough to merge into the psieg/Lightpack repo.

I think there are some inherent problems to scale this up to multiple Philips Hue bulbs (increased latency). I only have one bulb in this room, so I will not (can not) implement support for more than one bulb.

However I've read something about a new API which is faster and also handles simultaniously setting multiple bulbs better. I'm not sure where and how to access this API, though. 

My code is here: https://github.com/TJ-91/Lightpack/tree/ftr_sync_philips_hue

The settings have to be done in the profile in a new chapter "[HueLights]" where at least the "URL" value has to be set: 

> [HueLights]
> URL=http://**<ip_to_hue_bridge>**/api/**<user>**/lights/**<light_number>**/state

Since the Philips Hue system only allows for roughly 10 requests per second, it will lag behind a little but it's decent imo.

Also the CPU consumption of Prismatik has gone up a bit. I'd be happy for some advice since I'm really not experienced in Qt. It took some time to figure out how to do the threading thing for the HueWorker class. However since I've replaced the "busy-wait-loop" with a more reasonable approach of slots and signals that thread likely is unnecessary now.

So currently it's more or less in a state of "it works well enough for me" and some design decisions are based on what's convenient and not what's the best design.

**Comment by yebo29** [comment]
Another option is to turn off lights whenever you enable the light pack

**Comment by psieg** [comment]
Thanks! I don't have one though, so I won't try this or take ownership. Anyone interested in this please let me know.

**Comment by Yaro96** [comment]
> 
> 
> Thanks! I don't have one though, so I won't try this or take ownership. Anyone interested in this please let me know.

I agree that it would be a nice feature inside Prismatik, philips have it's own app but running two screen capturing software at the same time is not ideal
